tasks/prepare.yml

```diff
@@ -66,10 +67,8 @@
     k0s_delegate_to: "{{ ansible_host }}"
     k0s_local_tmp_directory: "{{
         lookup('env', 'CUSTOM_LOCAL_TMP_DIRECTORY') |
-        default('/var/cache/ansible/k0s', true) }}/{{ k0s_main_version }}"
-  delegate_to: localhost
+        default('~/.cache/ansible/k0s', true) }}/{{ k0s_main_version }}"
```

Comment:
Why do you change a default when you can configure it via an environment variable?

Reply:
Well that was a quick and dirty way to make the role support direct download, not intended for a PR. The actual problem is the delegate_to: localhost, as the desired outcome is to change the variable k0s_delegate_to: "{{ ansible_host }}". As currently written it will update host_vars['localhost']['k0s_delegate_to'] and not the appropriate ansible_host its running on.